src/components/Notifications/TimeConfig.tsx

```diff
@@ -45,14 +51,15 @@ export const TimeConfigComponent: React.FunctionComponent = () => {
         setShowCustomSelect(true);
     }, []);
 
-    const handleTimeSelect = React.useCallback(() => {
+    const handleTimePrefSelect = React.useCallback(() => {
+        setTimeSelect(timeSelect);
         setIsOpen(false);
-    }, []);
+    }, [ timeSelect ]);
 
-    const handleButtonSave = React.useCallback((time) => {
+    const handleButtonSave = React.useCallback((timeSelect) => {
```

Comment:
```suggestion
    const handleButtonSave = React.useCallback(() => {
```

Remove `timeSelect`, it's not a parameter sent to this function. This function only receives an event.
You want to use `timeSelect` from the outer scope.

Reply:
You will need to add `timeSelect` to the dependencies of ths `useCallback` function:
https://github.com/RedHatInsights/notifications-frontend/blob/f75e16d225d4ce0b9759f818aab8678bf3f1ddf3/src/components/Notifications/TimeConfig.tsx#L74